### PR TITLE
Adjust auto-import sorting to better match isort

### PR DESCRIPTION
**Pull request by cdce8p**
TypeScript and isort handle sorting of `_` differently. This can result in the auto-import insert position not matching the isort behavior.

This PR adds a workaround to better handle underscores when sorting import names.

#### Example
```py
from lib import (
    ...
    # auto-import
    MY_CONST_VAR,
    # isort
    ....
)

MYA_CONST_VAR  # import from lib
```

### CODE REVIEW

**Comment by erictraut** [comment]
Thanks for the contribution, @cdce8p. The change looks good to me.

@jakebailey, could you please give it a quick review as well?

File: packages/pyright-internal/src/analyzer/importStatementUtils.ts [hunk 1]
```diff
@@ -160,10 +160,16 @@ export function getTextEditsForAutoImportSymbolAddition(
             // Insert new symbol by import symbol type and then alphabetical order.
             // Match isort default behavior.
             const symbolNameType = _getImportSymbolNameType(symbolName);
+            // isort will prefer '_' over alphanumerical chars
+            // This can't be reproduced by a normal string compare in TypeScript, since '_' > 'A'.
+            // Replace all '_' with '=' which guarantees '=' < 'A'.
+            // Safe to do as '=' is an invalid char in Python names.
+            const symbolNameCompare = symbolName.replace('_', '=');
```
Comment (jakebailey) [inline packages/pyright-internal/src/analyzer/importStatementUtils.ts:167]:
```suggestion
            const symbolNameCompare = symbolName.replace(/_/g, '=');
```

I'm thinking you intended this to replace all underscores? replace only replaces the first match unless you use a regex. (We can't use replaceAll yet, so this is the right way to replace all.)

Comment (cdce8p) [reply packages/pyright-internal/src/analyzer/importStatementUtils.ts]: 👍🏻 Updated the code and added an additional test case for it.

Comment (erictraut) [reply packages/pyright-internal/src/analyzer/importStatementUtils.ts]: Good catch @jakebailey!

**Pull request merged**